controller/lib/core/comms.cpp

```diff
@@ -1,198 +1,31 @@
 #include "comms.h"
+
+#include "alg.h"
 #include "hal.h"
 #include "network_protocol.pb.h"
-#include "serdes.h"
+#include <pb_common.h>
+#include <pb_decode.h>
+#include <pb_encode.h>
 
-#define PACKET_LEN_MAX (32)
-static uint8_t tx_buffer[PACKET_LEN_MAX];
-static uint16_t tx_data_length;
-static bool output_buffer_ready = false;
+// TODO: Change last_{tx,rx}_ms to uint64 once Hal.millis() is uint64_t.
+//
+static uint8_t tx_buffer[ControllerStatus_size];
+static uint8_t tx_idx = 0;
+static uint16_t tx_bytes_remaining = 0;
+static uint32_t last_tx_ms = -1;
```

Comment:
It looks very icky to have an unsigned type initialized as -1. Can it be initialized to 0?

Reply:
That was how I wrote it initially, and it's actually a bug that I caught with my test.  Added a comment explaining the bug.